prog/test/github_runner.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+require "octokit"
+require "yaml"
+
+class Prog::Test::GithubRunner < Prog::Test::Base
```

Comment:
What do you think about splitting this prog into two, similar to the VmGroup and Vm? The current version of `Prog::Test::GithubRunner` maintains a list of test cases and manages all workflow cases. We could move the operations for individual workflow cases to another prog, while the parent program simply waits for the child progs to complete. The names are just examples.

- `Prog::Test::GithubRunner`: It's the parent prog. We assemble this one. It buds `Prog::Test::GithubWorkflow` for each test cases, and waits for children be done.
- `Prog::Test::GithubWorkflow`: It manages steps for a single test case. The current `trigger_test_run` and `test_run_conclusion` helpers are actually a label in it. It triggers and checks only the specified test case, without iterating over multiple cases. Additionally, it downloads the image for the given test case, enabling to push instead of bud `Prog::DownloadBootImage`

Reply:
It also helps us track each test case separately and identify the failed ones more easily.

Reply:
I find separating vm and vmgroup useful as we are doing multiple tests in vm after creating it. Having them in the vmgroup would make it more complex. 

Though, in that case we only trigger the run and check the result. If we implement it in the way you mentioned I would still have loops over all the child strands to check and cancel them. I also find propagating failure from child to parent as overhead. So, I'll be keeping it as it is.